project_mis_report_budget/views/mis_budget_views.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="project_mis_budget_form_view" model="ir.ui.view">
+        <field name="name">mis.budget.form</field>
+        <field name="model">mis.budget</field>
+        <field name="inherit_id" ref="mis_builder_budget.mis_budget_form_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='report_id']" position="after">
+                <field name="is_group_project_user" invisible="1"/>
+                <field name="project_id" attrs="{'readonly': [('is_group_project_user', '=', False)]}"/>
```

Comment:
Not readonly, but invisible (if a user does not belong to Project Users, he won't have read access indeed):
```suggestion
                <field name="project_id" groups="project.group_project_manager"/>
```

And, then, if `is_group_project_user` field is no longer used for this model, it could be removed.

Reply:
Ouch, when I've edited suggested code, I made a mistake: referred to Project Manager group, but Project Users group is enough, sorry!